compiler/commands.nim

```diff
@@ -444,7 +466,7 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
   of "backend", "b":
     let backend = parseEnum(arg.normalize, TBackend.default)
     if backend == TBackend.default: localError(conf, info, "invalid backend: '$1'" % arg)
-    conf.backend = backend
+    handleBackend(conf, backend)
```

Comment:
this prevents doing:
`nim r -b:c -b:cpp main`
this is useful to provide a way to override a backend in an command line, eg:
you have a command line template `nim r -b:c` and then you simply add a flag to override via `nim r -b:c -b:cpp`

right now there's a bug where this doesn't work (I can fix it unless you want to fix it), but your patch makes it worse because `handleBackend` has side effect.

TLDR: parsing the flag `--backend` should have zero side effect beyond setting `conf.backend = backend`, so that it can be overridden in subsequent flags

Reply:
This is definitely wrong:
  parsing the flag --backend should have zero side effect beyond setting conf.backend = 
  backend

Reply:
it's by design; the side effects are deferred until the command is processed (eg `nim doc` etc).

I'm fixing this issue + others in https://github.com/nim-lang/Nim/pull/14363 which should supersede your PR